content/push/device.textile

```diff
@@ -0,0 +1,392 @@
+languages:
+  - android
+  - swift
+  - objc
+redirect_from:
+  - /realtime/push/activate-subscribe
+  - /general/versions/v1.1/push/activate-subscribe
+  - /general/push/activate-subscribe
+---
+
+To push notifications with Ably, you must first configure the device using its platform-specific notification service (APNs for iOS or FCM for Android) and integrate it with Ably's infrastructure. You can then activate the push notifications service, either directly or via a server.
+
+h3(#configure). Configure devices
+
+Configuration is the first step in setting up push notifications with Ably. This step requires setting up the necessary infrastructure on the device's operating system or platform, such as iOS with "APNs":https://developer.apple.com/notifications/ or Android with "FCM":https://firebase.google.com/docs/cloud-messaging, as well as on the Ably platform.
+
+<a href="@content/diagrams/push-process-2.png" target="_blank">
+  <img src="@content/diagrams/push-process-2.png" style="width: 100%" alt="Push Notifications in Ably">
+</a>
+
+h3. Configure an Android device
+
+* Go to the "Firebase Console.":https://firebase.google.com/
+* Click *add project* and follow the steps to *create and manage service account keys*.
+* Download your Service account key file containing your Privacy Enhanced Mail (PEM) cert and PEM private key.
+* In your Ably "dashboard":https://ably.com/accounts, navigate to the *notifications tab* under your app settings.
+* Go to *push notifications setup*, click *configure push*.
+* Add your *PEM cert* and *PEM private key*.
+
+h3. Configure an iOS device
+
+* Go to the "Apple Developer Program.":https://developer.apple.com/programs/
+* Use the Apple Developer portal to create a *push notification* service certificate for your app.
+* Export the certificate as a *.p12* file.
+* In your Ably "dashboard":https://ably.com/accounts, navigate to the *notifications tab* under your app settings.
+* Go to *push notifications setup*, click *configure push*.
+* Add your *.p12* file.
+
+h3. Integrate Ably into your app
+
+The following steps guide you through the "Ably SDK":https://ably.com/docs/sdks integration process:
+
+```[android]
+// Define the custom action for push notifications
+public class AblyPushNotificationService extends FirebaseMessagingService {
+    public static final String PUSH_NOTIFICATION_ACTION = AblyPushNotificationService.class.getName() + ".PUSH_NOTIFICATION_MESSAGE";
+
+    // This method is overridden to handle any messages received while the app is in the foreground or background.
+    @Override
+    public void onMessageReceived(@NonNull RemoteMessage message) {
+        // Process the incoming message here
+    }
+
+    //This method is overridden to handle token refresh, ensuring that Ably is always updated with the latest token.
+    @Override
+    public void onNewToken(@NonNull String s) {
```

Comment:
I suggest using `token` instead of `s`, I think this way code looks more professional